### DISCUSSION TITLE
DEX: allow double swaps in a single step

### DISCUSSION THREAD

**Discussion by nippur72** [DeFiCh/app#857] [Ideas]
Currently the DEX allows you to swap only from and to `DFI`, so if you want to i.e. swap `BTC` for `ETH` you need first to swap `BTC-DFI` and then `DFI-ETH`, incurring in double fee and slippage (because the first swap takes time to confirm on the blockchain and the price may move in the while).

My idea is to allow the swap in a single step in the same transaction; under the hood it would be two swaps involving two different pools, but quicker than if you do it manually. 

If implemented, that would multiply arbitrage opportunities among all pairs of coins, increasing traffic in the blockchain and attracting users who are not interested in `DFI` or don't want to be exposed to it.

What do you think about it?

**Comment by John-Gee** [comment]
Does it actually need to be 2 transactions? Couldn't we have these pools as well (but with rewards still paid in DFI obviously)?

But yeah, I like it!

**Reply by nippur72** [reply]
dedicated pools would be ideal, but currently the main incentive for putting coins into a pool comes from the minted block rewards. So new rewards should be created for the new pools, which I don't think it's likely or even desirable: the APR of existing pools would decrease, and you would actually reward people for not holding DFI (like in a BTC-ETH liquidity pool).

**Reply by John-Gee** [reply]
Fair points.
